pints/_mcmc/__init__.py

```diff
@@ -301,24 +303,54 @@ def __init__(
             self, log_pdf, chains, x0, sigma0=None, transformation=None,
             method=None):
 
-        # Check function
-        if not isinstance(log_pdf, pints.LogPDF):
-            raise ValueError('Given function must extend pints.LogPDF')
+        if isinstance(log_pdf, pints.LogPDF):
+            self._multi_logpdf = False
+
+        else:
+            self._multi_logpdf = True
+            try:
+                log_pdfs_iter = iter(log_pdf)  # noqa
+            except TypeError:
+                raise ValueError('`log_pdf` must either extend pints.LogPDF, '
+                                 'or be a list of objects which extend '
+                                 'pints.LogPDF')
+
+            if len(log_pdf) != chains:
+                raise ValueError('`log_pdf` must either extend pints.LogPDF, '
```

Comment:
This is a valueerror. As are the remaining ones. (The contents of the right type arg are of the wrong type = a value error, imo)